lib/WebService/PayPal/PaymentsAdvanced/Error/Role/HasHTTPResponse.pm

```diff
@@ -0,0 +1,30 @@
+package WebService::PayPal::PaymentsAdvanced::Error::Role::HasHTTPResponse;
+
+use Moo::Role;
+
+use Types::Standard qw( InstanceOf Int );
+use Types::URI qw( Uri );
+
+has http_status => (
+    is       => 'ro',
+    isa      => Int,
+    required => 1,
+);
+
+has http_response => (
+    is       => 'ro',
+    isa      => InstanceOf ['HTTP::Response'],
+    required => 1,
+);
+
+has request_uri => (
+    is       => 'ro',
+    isa      => Uri,
+    coerce   => 1,
+    required => 1,
+);
```

Comment:
We could get it from the response, although we would have to set that in the test. We would probably want some sort of check to make sure the response has it too so that future people don't get confusing message due to `request` being undefined in some circumstance.